src/test/java/com/example/javaee_jpa_hibernate/test_helper/InvoiceGenerator.java

```diff
@@ -0,0 +1,112 @@
+package com.example.javaee_jpa_hibernate.test_helper;
+
+import com.example.javaee_jpa_hibernate.model.Invoice;
+import com.example.javaee_jpa_hibernate.model.counterparty.Address;
+import com.example.javaee_jpa_hibernate.model.counterparty.Counterparty;
+import com.example.javaee_jpa_hibernate.model.invoice_item.InvoiceItem;
+import com.example.javaee_jpa_hibernate.model.invoice_item.Vat;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+public class InvoiceGenerator {
+    public static Address generateAddress() {
+        return new Address("97-420", "Lodz", "Piotrkowska", "34");
+    }
+    public static Counterparty generateCounterparty() {
+        Counterparty counterparty = CounterpartyBuilder.builder()
+                .withCompanyName("LPT")
+                .withAddress(generateAddress())
+                .withPhoneNumber("345678912")
+                .withNIP("12345678")
+                .withBankName("PKO")
+                .withBankNumber("2345")
+                .build();
+        return counterparty;
+    }
+
+    public static Counterparty generateCounterparty(String companyName, Address address, String phoneNumber,
```

Comment:
There is not much sense in having such methods - you have a really nice builder, but wrapped it with a method that needs 4 String parameters in a row!